### PR TITLE
Port remove weakrefernce and optimise applySettings

### PR DESCRIPTION
**Pull request by Chaoba**
## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-kotlin-binary-compatibility` CI will fail.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Mapbox welcomes participation and contributions from everyone.
 
 # main
+## Features ✨ and improvements 🏁
+* Optimize how plugins handle settings changes. Call `applySettings` only when settings value changes. ([#1189](https://github.com/mapbox/mapbox-maps-android/pull/1189))
+* Add new Exception type `MapboxMapMemoryLeakException` which will be thrown when there is a leak for nativeMap. ([1193](https://github.com/mapbox/mapbox-maps-android/pull/1193))
```
Comment (alexander-kulikovskii) [inline CHANGELOG.md:8]:
```suggestion
* Add new exception type `MapboxMapMemoryLeakException` which will be thrown when there is a leak for nativeMap. ([1193](https://github.com/mapbox/mapbox-maps-android/pull/1193))
```

File: sdk/src/main/java/com/mapbox/maps/Style.kt [hunk 2]
```diff
@@ -21,9 +20,17 @@ import java.nio.ByteBuffer
  * @property pixelRatio the scale ratio of the style, default the device pixel ratio
  */
 class Style internal constructor(
-  private val styleManagerRef: WeakReference<StyleManagerInterface>,
+  private val styleManagerRef: StyleManagerInterface,
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:23]:
```suggestion
  private val styleManager: StyleManagerInterface,
```

File: sdk/src/main/java/com/mapbox/maps/Style.kt [hunk 3]
```diff
@@ -21,9 +20,17 @@ import java.nio.ByteBuffer
...
   override val pixelRatio: Float
 ) : StyleInterface {
+
+  private fun getStyleManager(): StyleManagerInterface {
+    if (styleManagerRef == null) {
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:28]: sorry, maybe I missed smth, but why uninitialized field throw `MapboxMapMemoryLeakException`?

**Comment by pengdev** [comment]
@alexander-kulikovskii @Chaoba I think we should probably address these review comments in the main branch and cherry-pick in v10.4 release branch.

**Review by yunikkk — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/Style.kt (hunk 3 above)
Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/Style.kt:28]: Good catch. we need to reset this reference in onDestroy.

**Pull request closed without merging**